libfprint/drivers/goodixtls/5395/goodix_device.c

```diff
@@ -258,3 +265,114 @@ gboolean fpi_goodix_device_reset(FpDevice *dev, guint8 reset_type, gboolean irq_
     GError *error = NULL;
     return fpi_goodix_device_send(dev, message, TRUE, 500, FALSE, &error);
 }
+
+gboolean fpi_goodix_device_gtls_connection(FpDevice *dev, GError *error)
+{
+    fp_dbg("fpi_goodix_device_gtls_connection()");
+    fp_dbg("Starting GTLS handshake");
+    FpiGoodixDevice *self = FPI_GOODIX_DEVICE(dev);
+    FpiGoodixDevicePrivate *priv = fpi_goodix_device_get_instance_private(self);
+    priv->gtls_params = fpi_goodix_device_gtls_init_params();
+    fpi_goodix_device_gtls_client_hello_step(dev);
+    if (!fpi_goodix_device_gtls_server_identity_step(dev, error))
+    {
+        return FALSE;
+    }
+    // fpi_goodix_gtls_server_done_step();
+    fp_dbg("GTLS handshake successful");
+    return TRUE;
+}
+
+GoodixGTLSParams *fpi_goodix_device_gtls_init_params(void)
```

Comment:
I think we have to move gtls related logic to another .c file. (example: fpi_gtls or fpi_goodix_gtls

Reply:
I know it. After our discussion on discord, I already moved this code and the following into a dedicated file.
Now I'm working on the HMAC creation phase, I hope that I'll push the update this night.
Thanks for the review